### PR TITLE
Ensure interior empty list items and some nbsp are not stripped away

### PR DESCRIPTION
**Pull request by christhekeele**
The first commit here adds a Gemfile and rspec to the gemspec for easier development, and reworks the test suite with some custom rspec helpers to remove boilerplate, and makes use of the indentation-aware `<<~HEREDOC` to add some consistency.

It changes neither library code nor spec code.

The second commit contains specs written when trying to repro and debug small errors observed by customers during editing experiences with TinyMCE, hence the name `tinymce_examples_spec`.

- interior empty `<li>`s are no longer stripped
- certain `&nbsp;` are no longer stripped away
- adds some tests for extra percents in output from span quicktags fixed by jeremy previously
- adds some tests for probing around issues with whitespace in tables

### CODE REVIEW

File: spec/tinymce_examples_spec.rb [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+  context "edge case with nested formats with spans and &nbsp;" do
+
+    it "should handle nested formatting" do
+      html = <<~HTML
+        test<i>test<b>test</b></i>
+      HTML
+
+      markup = <<~MARKUP
+        test_test*test*_
```
Comment (k1w1) [inline spec/tinymce_examples_spec.rb]: Is this markup correct? When I put that markup into JIRA it is not formatted, I still see the * and _.

File: spec/tinymce_examples_spec.rb [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+      expect(html).to match_markup(markup)
+    end
+
+    it "should handle trailing %nbsp; inside formats" do
+      html = <<~HTML
+        <b>test&nbsp;</b>
+      HTML
+
+      markup = <<~MARKUP
+        *test *
```
Comment (k1w1) [inline spec/tinymce_examples_spec.rb]: This also does not give the result I expect in JIRA. I see the * instead of the text being bold.

File: spec/tinymce_examples_spec.rb (hunk 1 above)
Comment (christhekeele) [inline spec/tinymce_examples_spec.rb]: It looks like the correct way to do this is `test{_}test{*}test*_`.

File: spec/tinymce_examples_spec.rb (hunk 2 above)
Comment (christhekeele) [inline spec/tinymce_examples_spec.rb]: The only way I can get this formulation (generated by tinymce when applying formatting to highlighted text or nested formats across words, IIRC) to work in JIRA is something like: `*test&nbsp;{*}test`. TinyMCE does give us the nbsp in these contexts but we throw it away for a normal space.

Alternatives like shifting the padding spaces outside the tags would force an extra auto-update loop between JIRA and Aha! to normalize.
